lib/ddtrace/contrib/redis/patcher.rb

```diff
@@ -80,14 +80,19 @@ def call_pipeline(*args, &block)
               return call_pipeline_without_datadog(*args, &block) unless pin && pin.tracer
 
               response = nil
-              pin.tracer.trace(Datadog::Contrib::Redis::Ext::SPAN_COMMAND) do |span|
-                span.service = pin.service
-                span.span_type = Datadog::Contrib::Redis::Ext::TYPE
-                commands = args[0].commands.map { |c| Datadog::Contrib::Redis::Quantize.format_command_args(c) }
-                span.resource = commands.join("\n")
-                Datadog::Contrib::Redis::Tags.set_common_tags(self, span)
-                span.set_metric Datadog::Contrib::Redis::Ext::METRIC_PIPELINE_LEN, commands.length
+              commands = args[0].commands.map { |c| Datadog::Contrib::Redis::Quantize.format_command_args(c) }
+
+              if commands.any?
```

Comment:
So this is interesting because it raises the question that if there are no commands, but the pipeline still runs, do we want to instrument it? (There might be a case for either one. I can imagine you might want to see that the Redis pipeline ran, even if it didn't do anything.)

If no, then I think this change is fine. If yes, then maybe we should give the resource name something else that isn't blank (e.g. `null`, `(none)`, etc.).

Thoughts?

Reply:
@delner Agree. I thought about it before and it seemed like a piece of useless information but at the moment it makes more sense. Understanding that your pipeline doesn't make any commands is a great insight into a transaction part.

I'll change the implementation.